docs/apis/communication.md

```diff
@@ -2,47 +2,345 @@
+  "messageOpinionTriggered": true,
+  "timestampOpinion": "Like",
+  "timestampLoK": "LevelOfKnowledge(Two)"
+}
+```
+
+#### Results
+|Return field | Type | Description|
+|:-----|:------|:------|
+| `id`  | `string` | Message ID. |
+| `opinionFormedTime`  | `int64` | Time when the node formed full opinion. |
+| `payloadOpinionFormed`  | `bool` | Flag indicating whether the node formed opinion about the payload. |
+| `timestampOpinionFormed`  | `bool` | Flag indicating whether the node formed opinion about the timestamp. |
+| `messageOpinionFormed`  | `bool` | Flag indicating whether the node formed opinion about the message. |
+| `messageOpinionTriggered`  | `bool` | Flag indicating whether the node triggered an opinion formed event for the message. |
+| `timestampOpinion`  | `string` | Opinion about the message's timestamp. |
+| `timestampLoK`  | `bool` | Level of knowledge about message's timestamp. |
+| `error`   | `string` | Error message. Omitted if success.    |
+
+
+## `/data`
+A data message is simply a `Message` containing some raw data (literally bytes). This type of message has therefore no real functionality other than that it is retrievable via `GetMessage`.
+
+### Parameters
+
+| **Parameter**            | `payload`      |
+|--------------------------|----------------|
+| **Required or Optional** | required       |
+| **Description**          | data bytes   |
+| **Type**                 | base64 serialized bytes         |
+
+
+#### Body
+
+```json
+{
+  "payload": "dataBytes"
+}
+```
+
+### Examples
+
+#### cURL
+
+```shell
+curl --location --request POST 'http://localhost:8080/data' \
+--header 'Content-Type: application/json' \
+--data-raw '{"payload": "dataBytes"}'
+```
+
+#### Client lib
+
+##### `Data(data []byte) (string, error)`
+```go
+messageID, err := goshimAPI.Data([]byte("Hello GoShimmer World"))
+if err != nil {
     // return error
 }
+```
+Note that there is no need to do any additional work, since things like tip-selection, PoW and other tasks are done by the node itself.
 
-// will print "Hello GoShimmer World"
-fmt.Println(string(message.Payload))
+### Response examples
+```json
+{
+  "id": "messageID" 
+}
 ```
 
-Note that we're getting actual `Message` objects from this call which represent a vertex in the communication layer Tangle. It does not matter what type of payload the message contains, meaning that `FindMessageByID` will also return messages which contain value objects or DRNG payloads.
+### Results
+|Return field | Type | Description|
+|:-----|:------|:------|
+| `id`  | `string` | Message ID of the message. Omitted if error. |
+| `error`   | `string` | Error message. Omitted if success.    |
+
+
+
 
-#### Send Payload
+## `/messages/payload`
 `SendPayload()` takes a `payload` object of any type (data, value, drng, etc.) as a byte slice, issues a message with the given payload and returns its `messageID`. Note, that the payload must be valid, otherwise an error is returned.
```

Comment:
```suggestion
`SendPayload()` takes a `payload` object of any type (data, transaction, drng, etc.) as a byte slice, issues a message with the given payload and returns its `messageID`. Note, that the payload must be valid, otherwise an error is returned.
```